spikingjelly/activation_based/triton_kernel/flex_sn_inductor/custom_ops.py

```
@@ -183,18 +343,112 @@ def _flexsn_inductor_training_impl(
+    *args: torch.Tensor,
+) -> tuple[torch.Tensor, ...]:
+    x_example = args[0]
+    T = x_example.shape[0]
+    NCL = x_example[0].numel()
+    dtype = x_example.dtype
+    output_seqs = [torch.empty_like(x_example) for _ in range(info.num_outputs)]
+    final_states = [_materialize_empty_from_spec(spec) for spec in state_template_specs]
+    saved_tensors = [
+        torch.empty_like(x_example) for _ in _saved_non_output_indices(info)
+    ]
+    grid = lambda meta: ((NCL + meta["BLOCK_NCL"] - 1) // meta["BLOCK_NCL"],)
```

Ruff (0.15.12)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `grid` as a `def`

(E731)